### PR TITLE
feat(aws-kms): support waiting period

### PR DESCRIPTION
**Pull request by hassanazharkhan**
Closes [#12218](https://github.com/aws/aws-cdk/issues/12218)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 1]
```diff
@@ -330,6 +330,22 @@ export interface KeyProps {
...
+  /**
+   * Specifies the number of days in the waiting period before
+   * AWS KMS deletes a CMK that has been removed from a CloudFormation stack.
+   *
+   * When you remove a customer master key (CMK) from a CloudFormation stack, AWS KMS schedules the CMK for deletion
+   * and starts the mandatory waiting period. The PendingWindowInDays property determines the length of waiting period.
+   * During the waiting period, the key state of CMK is Pending Deletion, which prevents the CMK from being used in
+   * cryptographic operations. When the waiting period expires, AWS KMS permanently deletes the CMK.
+   *
+   * Enter a value between 7 and 30 days.
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-kms/lib/key.ts:343]: There should be an error added if the value provided is not in the allowed range.

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 2]
```diff
@@ -330,6 +330,22 @@ export interface KeyProps {
...
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-key.html#cfn-kms-key-pendingwindowindays
+   * @default - 30 days
+   */
+  readonly pendingWindow?: number;
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-kms/lib/key.ts:348]: Use the `Duration` class here instead of a number.

**Review by NetaNir — DISMISSED** [review]

**Comment by hassanazharkhan** [comment]
@NetaNir PR now ready for another look

File: packages/@aws-cdk/aws-kms/lib/key.ts [hunk 3]
```diff
@@ -400,11 +416,19 @@ export class Key extends KeyBase {
       }
     }
 
+    const pendingWindowInDays = props.pendingWindow?.toDays();
+    if (pendingWindowInDays) {
+      if (pendingWindowInDays < 7 || pendingWindowInDays > 30) {
+        throw new Error(`'pendingWindow' value must between 7 and 30 days. Received: ${pendingWindowInDays}`);
+      }
+    }
+
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-kms/lib/key.ts:425]:
```suggestion
    let pendingWindowInDays;
    if (props.pendingWindow) {
      pendingWindowInDays = props.pendingWindow.toDays();
      if (pendingWindowInDays < 7 || pendingWindowInDays > 30) {
        throw new Error(`'pendingWindow' value must between 7 and 30 days. Received: ${pendingWindowInDays}`);
      }
    }

```

File: packages/@aws-cdk/aws-kms/README.md [hunk 4]
```diff
@@ -21,6 +21,17 @@ new kms.Key(this, 'MyKey', {
 });
 ```
 
+Define a KMS key with waiting period:
```
Comment (NetaNir) [inline packages/@aws-cdk/aws-kms/README.md:24]: Im not sure that we need to add this to the README of the module, I think the property documentation should be sufficient. I'll leave it to you to decide.

**Review by njlynch — APPROVED** [review]

**Pull request merged**